pkg/lockfile/parse-pnpm-lock.go

```diff
@@ -65,14 +65,27 @@ func startsWithNumber(str string) bool {
 
 // extractPnpmPackageNameAndVersion parses a dependency path, attempting to
 // extract the name and version of the package it represents
-func extractPnpmPackageNameAndVersion(dependencyPath string) (string, string) {
+func extractPnpmPackageNameAndVersion(dependencyPath string, lockfileVersion float64) (string, string) {
 	// file dependencies must always have a name property to be installed,
 	// and their dependency path never has the version encoded, so we can
 	// skip trying to extract either from their dependency path
 	if strings.HasPrefix(dependencyPath, "file:") {
 		return "", ""
 	}
 
+	if lockfileVersion == 9.0 {
```

Comment:
Can you add some comments inline for describing how 9.0 differs here?

Reply:
I'm not really sure how I can without heavily describing the actual format itself, which frankly I don't know for sure since there isn't a spec yet - I think anyone that's wanting to review this code should do so side-by-side with the fixtures to be able to see how the different versions differ

Reply:
Can we just include some examples inline in the comments here? 

.e.g. 

```
pnpm v9 differs by adopting the following format:
  @foo@1.0.0
  @foo/bar@1.0.0
 @foo@https:/....
```